firestore/README.md

```diff
@@ -227,7 +243,155 @@ FirestoreRecyclerAdapter adapter = new FirestoreRecyclerAdapter<Chat, ChatHolder
+an adapter that loads a generic `Item`:
+
+```java
+// The "base query" is a query with no startAt/endAt/limit clauses that the adapter can use
+// to form smaller queries for each page.  It should only include where() and orderBy() clauses
+Query baseQuery = mItemsCollection.orderBy("value", Query.Direction.ASCENDING);
+
+// This configuration comes from the Paging Support Library
+// https://developer.android.com/reference/android/arch/paging/PagedList.Config.html
+PagedList.Config config = new PagedList.Config.Builder()
+        .setEnablePlaceholders(false)
+        .setPrefetchDistance(10)
+        .setPageSize(20)
+        .build();
+
+// The options for the adapter combine the paging configuration with query information
+// and application-specific options for lifecycle, etc.
+FirestorePagingOptions<Item> options = new FirestorePagingOptions.Builder<Item>()
+        .setLifecycleOwner(this)
+        .setQuery(baseQuery, config, Item.class)
+        .build();
+```
+
+If you need to customize how your model class is parsed, you can use a custom `SnapshotParser`:
+
+```java
+...setQuery(..., new SnapshotParser<Item>() {
+    @NonNull
+    @Override
+    public Item parseSnapshot(@NonNull DocumentSnapshot snapshot) {
+        return ...;
+    }
+});
+```
+
+Next create the `FirestorePagingAdapter` object. You should already have a `ViewHolder` subclass
+for displaying each item. In this case we will use a custom `ItemViewHolder` class:
+
+```java
+FirestorePagingAdapter<Item, ItemViewHolder> adapter =
+        new FirestorePagingAdapter<Item, ItemViewHolder>(options) {
+            @NonNull
+            @Override
+            public ItemViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+                // Create the ItemViewHolder
+                // ,,,
+            }
+
+            @Override
+            protected void onBindViewHolder(@NonNull ItemViewHolder holder,
+                                            int position,
+                                            @NonNull Item model) {
+                // Bind the item to the view holder
+                // ...
+            }
+        };
+```
+
+Finally attach the adapter to your `RecyclerView` with the `RecyclerView#setAdapter()`.
+Don't forget to also set a `LayoutManager`!
+
+#### `FirestorePagingAdapter` lifecycle
+
+##### Start/stop listening
+
+The `FirestorePagingAdapter` listens for scrolling events and loads additional pages from the
+database only when needed.
+
+To begin populating data, call the `startListening()` method. You may want to call this
+in your `onStart()` method. Make sure you have finished any authentication necessary to read the
+data before calling `startListening()` or your query will fail.
+
+```java
+@Override
+protected void onStart() {
+    super.onStart();
+    adapter.startListening();
+}
+```
+
+Similarly, the `stopListening()` call removes the snapshot listener and all data in the adapter.
```

Comment:
> removes all data in the adapter

Is this true? Pretty sure all the data is just going to stay there. Unless you meant to call `submitList(Collections.emptyList())` in `stopListening()`?